be/src/util/arrow/starrocks_column_to_arrow.cpp

```diff
@@ -100,27 +141,16 @@ struct ColumnToArrowConverter<LT, AT, is_nullable, ConvDecimalGuard<LT, AT>> {
         return {high, low};
     }
 
-    static inline arrow::Status convert(const ColumnPtr& column, arrow::MemoryPool* pool,
-                                        std::shared_ptr<arrow::Array>& array) {
-        DCHECK(!column->is_constant() && !column->only_null());
-        std::unique_ptr<ArrowBuilderType> builder;
+    static inline arrow::Status convert(const ColumnPtr& column, int start_idx, int end_idx,
+                                        [[maybe_unused]] ColumnContext* column_context,
+                                        arrow::ArrayBuilder* array_builder) {
+        ARROW_RETURN_NOT_OK(check_const_and_only_null(column));
+        ArrowBuilderType* builder = down_cast<ArrowBuilderType*>(array_builder);
         if constexpr (is_nullable) {
             const auto* nullable_column = down_cast<NullableColumn*>(column.get());
             const auto* data_column = down_cast<StarRocksColumnType*>(nullable_column->data_column().get());
-            const auto* null_column = down_cast<NullColumn*>(nullable_column->null_column().get());
             const auto& data = data_column->get_data();
-            const auto num_rows = null_column->size();
-
-            if constexpr (lt_is_decimalv2<LT>) {
-                auto arrow_type = std::make_shared<ArrowType>(27, 9);
```

Comment:
why ignore decimal2?

Reply:
Here it's just used to get the builder. But now we use `arrow::MakeBuilder` in `ColumnToArrowArrayConverter::Visit` to get the builder, so this logic can be removed. 
```
ARROW_RETURN_NOT_OK(arrow::MakeBuilder(_pool, _arrow_type, &builder));
```